napari/layers/labels/_labels_utils.py

```diff
@@ -133,3 +137,68 @@ def get_dtype(layer):
         layer_dtype = type(layer_data_level)
 
     return layer_dtype
+
+
+def first_nonzero_coordinate(data, start_point, end_point):
+    """Coordinate of the first nonzero element between start and end points.
+
+    Parameters
+    ----------
+    data : nD array, shape (N1, N2, ..., ND)
+        A data volume.
+    start_point : array, shape (D,)
+        The start coordinate to check.
+    end_point : array, shape (D,)
+        The end coordinate to check.
+
+    Returns
+    -------
+    coordinates : array of int, shape (D,)
+        The coordinates of the first nonzero element along the ray, or None.
+    """
+    shape = np.asarray(data.shape)
+    length = np.linalg.norm(end_point - start_point)
+    length_int = np.round(length).astype(int)
+    coords = np.linspace(start_point, end_point, length_int + 1, endpoint=True)
+    clipped_coords = np.clip(coords, 0, shape - 1).astype(int)
+    nonzero = np.flatnonzero(data[tuple(clipped_coords.T)])
+    if len(nonzero) == 0:
+        return None
+    else:
+        return clipped_coords[nonzero[0]]
+
+
+def get_action_coordinate(layer, event):
```

Comment:
Can this maybe be `get_data_coordinate`? I find `action` quite confusing as this isn't an action it's getting - it's a vispy event